### PR TITLE
Dockerized development environment

### PR DESCRIPTION
**Pull request by jxstanford**
Review HACKING.md for setup instruction.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -1,288 +1,169 @@
-# Goldstone Hacking Guide
+Goldstone Server Hacking Guide
+================
 
-This explains how to install and run Goldstone locally, so you can do code development on it.
 
-The recommended steps are:
+This document explains how to install and run Goldstone Server locally (mostly in docker containers), so you can do code development on the project.  The instructions assume a MacOS X development environment with [homebrew](http://brew.sh/) and [Virtualbox](https://www.virtualbox.org/wiki/Downloads) installed.  The setup process for other environments will be similar, but inevitably be different.
```
Comment (lexjacobs) [inline HACKING.md]: Unnecessary?
"The setup process for other environments will be similar, but inevitably be different."

File: HACKING.md [hunk 2]
```diff
@@ -1,288 +1,169 @@
...
-Remember that an instance of Goldstone running locally can be used to monitor a
-local *or remote* OpenStack installation. The server on which Goldstone runs is
-independent from the OpenStack cloud's location.
+Prerequisites
+----------------
 
+Install various prerequisite packages:
 
-## Installation
+    $ brew upgrade
```
Comment (lexjacobs) [inline HACKING.md]: `$ brew update` ?

From `man brew` : 
update [--rebase]
              Fetch the newest version of Homebrew and all formulae from GitHub using git(1).

upgrade [install-options] [formulae]
              Upgrade outdated, unpinned brews.

**Comment by lexjacobs** [comment]
Just a couple of inline comments above and then :+1:

File: HACKING.md (hunk 1 above)
Comment (stugots) [inline HACKING.md]: The correct name for the operating system is OS X.

File: HACKING.md [hunk 3]
```diff
@@ -1,288 +1,169 @@
...
-
-```bash
-#!/bin/bash
-
+Configure VirtualBox Networking
+---------------------------------------
 
-echo "shutting down redis"
-pkill -f redis
+This section assumes that you will be using the pre-build RDO image with an IP address of 172.24.4.100.  If you are using a different image, you may need to adjust the network configuration to match your VM definition.
```
Comment (stugots) [inline HACKING.md]: "using the pre-_built_ RDO..."

File: HACKING.md [hunk 4]
```diff
@@ -1,288 +1,169 @@
...
+|es_9300_RDO|TCP|172.24.4.1|9300||9300|
+|es_9300_local|TCP|127.0.0.1|9300||9300|
+|logstash_syslog_RDO|TCP|172.24.4.1|5514||5514|
+|logstash_syslog_local|TCP|127.0.0.1|5514||5514|
+|logstash_metrics_RDO|UDP|172.24.4.1|5516||5516|
+|logstash_metrics_local|UDP|127.0.0.1|5516||5516|
+|postgres_RDO|TCP|172.24.4.1|5432||5432|
+|postgres_local|TCP|127.0.0.1|5432||5432|
+|redis_local|TCP|127.0.0.1|6379||6379|
+|ssh|TCP|127.0.0.1|2022||22|
```
Comment (stugots) [inline HACKING.md]: This table is broken. 

screen shot 2015-06-06 at 11 45 32 am

File: HACKING.md [hunk 5]
```diff
@@ -1,288 +1,169 @@
...
-```
+![enter image description here](https://lh3.googleusercontent.com/Hy1sDfWbYbLvhJjZa7kNSXXImGtri7zIlwPEazNwk3s=w797-h634-no)
 
-Edit your `pg_hba.conf` file.  If you installed with brew, this should be in `/usr/local/var/postgres/`.  See the [INSTALL file](INSTALL.md) for the modifications.
+Activate the Virtualenv 
+----------------------------------------------------
+Once all of the initial setup has been completed, you can activate the virtualenv by running the workon command. 
 
-After you've edited pg_hba.conf, reload postgres:
+> $ workon goldstone-server
```
Comment (stugots) [inline HACKING.md]: This line's formatting is different than the other bash lines. Delete the > and indent it four spaces.

File: HACKING.md [hunk 6]
```diff
@@ -1,288 +1,169 @@
...
+    $ pip install -r test-requirements.txt
+    $ pip install flower
 
-```bash
-$ service iptables restart
-```
+If the requirements files change, you should rerun the `pip install` commands.
 
-### Get the code
+*Note that the goldstone-server virtualenv is only meant to be run in a single terminal window. *
```
Comment (stugots) [inline HACKING.md]: This isn't bolded or emphasized in the output. Two **? Some other formatting problem?

File: HACKING.md [hunk 7]
```diff
@@ -1,288 +1,169 @@
-# Goldstone Hacking Guide
+Goldstone Server Hacking Guide
+================
```
Comment (stugots) [inline HACKING.md]: This file has a Github Markdown extension, but this is using ReST format controls.  Confusing to anyone editing the source, and if it doesn't confuse the markdown processor now, it eventually will.

File: HACKING.md (hunk 1 above)
Comment (jxstanford) [inline HACKING.md]: fixed

File: HACKING.md (hunk 2 above)
Comment (jxstanford) [inline HACKING.md]: fixed

File: HACKING.md (hunk 5 above)
Comment (jxstanford) [inline HACKING.md]: fixed

File: HACKING.md (hunk 7 above)
Comment (jxstanford) [inline HACKING.md]: fixed, was trying out a wysiwyg editor that led me a little astray.

File: HACKING.md (hunk 4 above)
Comment (jxstanford) [inline HACKING.md]: fixed

**Comment by jxstanford** [comment]
I think I addressed everything from review.

File: HACKING.md [hunk 8]
```diff
@@ -1,288 +1,167 @@
...
+    $ brew install boot2docker
+    $ brew install docker-compose
+    $ brew install pyenv-virtualenvwrapper
+    $ boot2docker init
 
-### Preliminaries
+## Fork and Clone Goldstone Repos
 
-If you're doing this on OS X, you need to install these packages:
+Depending on your contributor status (core or community), you will either create forks of the [goldstone-server](https://github.com/Solinea/goldstone-server) and [goldstone-docker](https://github.com/Solinea/goldstone-docker) Github repositories, or you will be working on on branches from the main repos.
```
Comment (stugots) [inline HACKING.md]: This offers a choice, either core or community status; then it says what community contributors should do. ("If you are a community contributor...")  Further instructions for core contributors are missing.  

Either fold everything together (which means inserting "your branch or fork" in multiple places) or put in specific instructions for core contributors.

File: HACKING.md (hunk 4 above)
Comment (stugots) [inline HACKING.md]: The table duplicates the screen snapshot. Will it eventually have additional columns?  If not, delete the table -- the snapshot makes the network configuration perfectly clear!

Comment (jxstanford) [inline HACKING.md]: I was thinking that it's pretty hard to cut/paste from an image, and if someone wanted to script the config, they would be able to quickly cut/paste from the table.  Also provides some protection from the image service being down.  Eventually, I think we can figure out some virtualbox commands to set this up automatically.  I'll delete it and we can shoot for the script when we get some slack.

Comment (stugots) [inline HACKING.md]: Ah, cutting and pasting. I didn't think of that. Good point! I retract my comment.

Comment (jxstanford) [inline HACKING.md]: undeleted!

**Comment by stugots** [comment]
:+1:

**Comment by lexjacobs** [comment]
:+1:

**Pull request merged**